File: DSRFQ.Web/Modules/ServerTypes/Master/DimensionUnitsColumns.ts
import { ColumnsBase, fieldsProxy } from '@serenity-is/corelib';
import { Column } from '@serenity-is/sleekgrid';
import { DimensionUnitsRow } from './DimensionUnitsRow';

export interface DimensionUnitsColumns {
    Id: Column<DimensionUnitsRow>;
    Code: Column<DimensionUnitsRow>;
    Name: Column<DimensionUnitsRow>;
    InsertDate: Column<DimensionUnitsRow>;
    InsertUserId: Column<DimensionUnitsRow>;
    UpdateDate: Column<DimensionUnitsRow>;
    UpdateUserId: Column<DimensionUnitsRow>;
    DeleteDate: Column<DimensionUnitsRow>;
    DeleteUserId: Column<DimensionUnitsRow>;
    IsActive: Column<DimensionUnitsRow>;
}

export class DimensionUnitsColumns extends ColumnsBase<DimensionUnitsRow> {
    static readonly columnsKey = 'Master.DimensionUnits';
    static readonly Fields = fieldsProxy<DimensionUnitsColumns>();
}

File: DSRFQ.Web/Modules/Master/DimensionUnits/DimensionUnitsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace DSRFQ.Master.Columns;

[ColumnsScript("Master.DimensionUnits")]
[BasedOnRow(typeof(DimensionUnitsRow), CheckNames = true)]
public class DimensionUnitsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Code { get; set; }
    public string Name { get; set; }
    public DateTime InsertDate { get; set; }
    public string InsertBy { get; set; }
    public DateTime UpdateDate { get; set; }
    public string UpdateBy { get; set; }
    public DateTime DeleteDate { get; set; }
    public string DeleteBy { get; set; }
}